src/Gitonomy/Git/Repository.php

```diff
@@ -516,6 +516,30 @@ public function setDescription($description)
         return $this;
     }
 
+    /**
+     * Executes pull command.
+     *
+     * @return Repository the current repository
+     */
+    public function pull()
+    {
+        $result = $this->run('pull');
+
+        return preg_match('/\d (?:file|files) changed, \d insertion\(\+\), \d deletion\(\-\)/', $result) === 1;
```

Comment:
`run` method ensures the process returns a 0 exit code, meaning it executed successfully.

If an error occurs, it should throw an exception, so there's no need to check the output.

You should simply return nothing, and the associated test will fail if an error is thrown.

Reply:
I think the idea here is to tell the difference between "nothing was pulled" and return false, and "some new stuff was pulled" and return true.

Reply:
@GrahamCampbell @alexandresalome is right about my intention.

@alexandresalome would a simple `$this->run() !== null` check suffice?

and than again, how would this affect the unit test, do we want to check the actual output, or just if the command ran with exit code 0?